sources/packages/web/src/components/aest/students/modals/BypassRestrictionModal.vue

```diff
@@ -0,0 +1,272 @@
+                {{ restrictionBypassDetails.removedBy }}
+              </v-col>
+            </v-row>
+          </v-card>
+        </template>
+      </template>
+      <template #footer>
+        <footer-buttons
+          :processing="loading"
+          primaryLabel="Bypass restriction"
+          @secondaryClick="cancel"
+          @primaryClick="bypassRestriction"
+          :showPrimaryButton="
+            !restrictionBypassDetails?.applicationRestrictionBypassId
+          "
+        />
+      </template>
+    </modal-dialog-base>
+  </v-form>
+</template>
+<script lang="ts">
+import { ApiProcessError, SelectItemType, VForm } from "@/types";
+import { ref, defineComponent, reactive, watchEffect, watch } from "vue";
+import {
+  useRules,
+  useModalDialog,
+  useFormatters,
+  useSnackBar,
+} from "@/composables";
+import {
+  ApplicationRestrictionBypassAPIOutDTO,
+  AvailableStudentRestrictionsAPIOutDTO,
+  BypassRestrictionAPIInDTO,
+  RestrictionBypassBehaviors,
+} from "@/services/http/dto";
+import ModalDialogBase from "@/components/generic/ModalDialogBase.vue";
+import { ApplicationRestrictionBypassService } from "@/services/ApplicationRestrictionBypassService";
+
+export default defineComponent({
+  components: {
+    ModalDialogBase,
+  },
+  emits: { restrictionBypassed: null },
+  props: {
+    applicationId: {
+      type: Number,
+      required: true,
+    },
+    applicationRestrictionBypassId: {
+      type: Number,
+      required: false,
+    },
+  },
+  setup(props, { emit }) {
+    const snackBar = useSnackBar();
+    const { dateOnlyLongString } = useFormatters();
+    const restrictionsToBypass = ref([] as SelectItemType[]);
+    const restrictionBypassDetails = ref(
+      {} as ApplicationRestrictionBypassAPIOutDTO,
+    );
+    const formModel = reactive({} as BypassRestrictionAPIInDTO);
+    const availableStudentRestrictions = ref(
+      {} as AvailableStudentRestrictionsAPIOutDTO,
+    );
+    const { showDialog, showModal, resolvePromise, loading } =
+      useModalDialog<boolean>();
+    const bypassRestrictionForm = ref({} as VForm);
+    const { checkNullOrEmptyRule, checkNotesLengthRule } = useRules();
+    const note = ref("");
+    const cancel = () => {
+      restrictionBypassDetails.value =
+        {} as ApplicationRestrictionBypassAPIOutDTO;
+      bypassRestrictionForm.value.reset();
+      resolvePromise(false);
+    };
+    const bypassRestriction = async () => {
+      const validationResult = await bypassRestrictionForm.value.validate();
+      if (!validationResult.valid) {
+        return;
+      }
+      try {
+        loading.value = true;
+        await ApplicationRestrictionBypassService.shared.bypassRestriction({
+          applicationId: props.applicationId,
+          studentRestrictionId: formModel.studentRestrictionId,
+          bypassBehavior: formModel.bypassBehavior,
+          note: formModel.note,
+        });
+        emit("restrictionBypassed");
```

Comment:
Usually, there is no need to emit an event because the code in the consumer can just await by the showModal.
Was there a reason this was needed to be implemented differently?